File: src/java/magmac/app/lang/java/JavaPost.java
package magmac.app.lang.java;

import magmac.app.compile.rule.NodeRule;
import magmac.app.compile.rule.Rule;
import magmac.app.compile.rule.StripRule;
import magmac.app.compile.rule.SuffixRule;
import magmac.app.compile.rule.TypeRule;
import magmac.app.lang.node.JavaValue;
import magmac.app.lang.node.Post;
import magmac.app.lang.node.PostVariant;

public final class JavaPost extends Post implements JavaFunctionSegmentValue {
    public JavaPost(PostVariant variant, JavaValue value) {
        super(variant, value);
    }

    public static Rule createPostRule(String type, String suffix, Rule value) {
        return new TypeRule(type, new StripRule(new SuffixRule(new NodeRule("child", value), suffix)));
    }

    public PostVariant variant() {
        return variant;
    }

    public JavaValue value() {
        return value;
    }
}

File: src/web/magmac/app/lang/node/JavaPost.ts
import { NodeRule } from "../../../../magmac/app/compile/rule/NodeRule";
import { Rule } from "../../../../magmac/app/compile/rule/Rule";
import { StripRule } from "../../../../magmac/app/compile/rule/StripRule";
import { SuffixRule } from "../../../../magmac/app/compile/rule/SuffixRule";
import { TypeRule } from "../../../../magmac/app/compile/rule/TypeRule";
import { JavaFunctionSegmentValue } from "../../../../magmac/app/lang/java/JavaFunctionSegmentValue";
import { Objects } from "../../../../java/util/Objects";
export class JavaPost {
	JavaPost(variant : PostVariant, value : JavaValue) : public {;;}
	createPostRule(type : String, suffix : String, value : Rule) : Rule {;;}
	variant() : PostVariant {;;}
	value() : JavaValue {;;}
}
